File: onyxsdk-pen/src/androidTest/java/com/onyx/android/sdk/pennative/PenInk.java
package com.onyx.android.sdk.pennative;

import android.graphics.Bitmap;

/** Test-only return object constructed by the installed Notable JNI library. */
public final class PenInk {
    private final float[] points;
    private final int[] pointSizeArray;
    private final Bitmap[] bitmaps;

    public PenInk(float[] points, int[] pointSizeArray, Bitmap[] bitmaps) {
        this.points = points;
        this.pointSizeArray = pointSizeArray;
        this.bitmaps = bitmaps;
    }

    public static PenInk create(float[] points, int[] pointSizeArray, Bitmap[] bitmaps) {
        return new PenInk(points, pointSizeArray, bitmaps);
    }

    public float[] getPoints() { return points; }
    public int[] getPointSizeArray() { return pointSizeArray; }
    public Bitmap[] getBitmaps() { return bitmaps; }
}

File: onyxsdk-pen/src/main/java/com/onyx/android/sdk/pennative/PenInk.kt
package com.onyx.android.sdk.pennative

import android.graphics.Bitmap

/** Immutable native ink payload compatible with `onyxsdk-pennative:1.0.4`. */
class PenInk(
    val points: FloatArray,
    val pointSizeArray: IntArray,
    val bitmaps: Array<Bitmap>,
) {
    override fun toString(): String = buildString {
        append("PenInk: ${points.size} -> [")
        if (points.isNotEmpty()) append(points.first())
        for (index in 1 until points.size) append(",${points[index]}")
        append(']')
    }

    companion object {
        @JvmStatic
        fun create(
            points: FloatArray,
            pointSizeArray: IntArray,
            bitmaps: Array<Bitmap>,
        ): PenInk = PenInk(points, pointSizeArray, bitmaps)
    }
}
